db/hints/manager.hh

```diff
@@ -152,6 +152,7 @@ public:
             seastar::scheduling_group _hints_cpu_sched_group;
             gms::gossiper& _gossiper;
             seastar::shared_mutex& _file_update_mutex;
+            uint64_t _total_replayed_segments_count = 0;
```

Comment:
I have a vague feeling that this is duplicating the replay_position functionality, which is a sort of logical barrier between commitlog entries.

Reply:
The replay_position could be used here, however we would also have to make sure that _segments_to_replay respects the order of replay_position (I don't think it does now).

Reply:
ASAIK replay_position defines the position of the currently replayed mutation in the current segment (a.k.a. a single commitlog file): see how it's created by the `commitlog::read_log_file(...)`.

While `_total_replayed_segments_count` counts the total number of replayed files.